### DISCUSSION TITLE
Is there a way for a custom interface field to get the input value from another field?

### DISCUSSION THREAD

**Discussion by jsx101** [directus/directus#16603] [Archive]
As per the title, I was wondering if it is possible to create a custom interface field that dynamically responds to the input value of another field in the same form. I've tried using useItems but it will only return saved items.

**Comment by br41nslug** [comment]
In a custom interface it should be possible to do this by "injecting" the form data altough this can be finicky `const values = inject('values');`

**Reply by jsx101** [reply]
Thanks, I'll give it a try!

**Reply by jsx101** [reply]
Hi @br41nslug, for inject(), I assume you mean this right https://vuejs.org/api/composition-api-dependency-injection.html? Can't seem to get it to work. I've tried using provide() in one custom interface and inject() in another as shown in the link, but I cannot pass a test string value between the components for these interfaces.

Has this been done before? Perhaps I'm doing this wrong, though I suppose it can't be done because both components do not share an ancestor/descendent relationship

**Reply by br41nslug** [reply]
The `provide` is done by the `v-form` and should be accessible from any component rendered by the `v-form` like interfaces. Using the snippet i shared above should get you all the values of this form.

**Reply by jsx101** [reply]
I just tested it out, and it works. Thank you so much!

**Comment by NigmaX** [comment]
Hi @jsx101 ive solved this by using the inject 
`inject: ['values'],`
and then use the watch to detect changes
```
  watch: {
    'values.value'(newVal, oldVal) {
     // do whatever you want here
    }
  },
```

**Reply by jsx101** [reply]
Thank you for your suggestion. I've tried it out but it still doesn't seem to work for me. 

Just to be sure I've gotten this right. This is the `extensions/interfaces` folder in my Directus project. I would like to pass a value from `interface-prototype-one` to `interface-prototype-two`
![Screenshot_20221127_091544](https://user-images.githubusercontent.com/40978303/204130070-2aa2ead5-5b2c-41dc-9702-89811b369078.png)

As a proof of concept to see if it can be done. This is the implementation for the Vue component in `interface-prototype-one`.
```
import { ref, provide } from 'vue';

...
   setup(props, { emit }) {
      const testStringRef = ref('test string ref');
      provide('testRef', testStringRef);
      provide('testString', 'test string');
   }
```


And this is the implementation for the Vue component in `interface-prototype-two`. I'm using Composition API but I assume I've written it correctly so it should work the same as your solution.
```
import { ref, inject} from 'vue';

...
   setup(props, { emit }) {
      const testRef = inject<Ref<string>>('testRef', ref('testRef not found'));
      const testString = inject<string>('testString', 'testString not found');
     
      watch(testRef, (testRefString, prevTestRefString)=>{
            console.log("Injected test string ref (in watch): " + testRefString);
            console.log("Injected test string (in watch): " + prevTestRefString);
      });

      console.log("Injected test string ref: " + testRef.value);
      console.log("Injected test string: " + testString);   }
```


However, this is the result I got in the console output
![image](https://user-images.githubusercontent.com/40978303/204130989-0856f0e3-941a-4040-8b43-5a40e00d63a5.png)

Any help on this is appreciated!

**Comment by w4-asad** [comment]
@jsx101 were you able to fix this issue? I'm currently on the exact same problem but nothing seems to work for me so far. Any help would be appreciated.

**Reply by jsx101** [reply]
Hi @w4-asad, I was able to fetch the value of other fields in the form, but am not able to pass any additional data from one field to another. You can inject values from other fields with the following code

```
const values = inject('values');
values.value['other_field_name']
```
